### PR TITLE
367 experiment registry

### LINKED ISSUE #367: Make __main__.wait_on_queue() more generic to handle different experiments
A set of plans appropriate for running in the `__main__` RunEngine should be able to be dynamically loaded and interpreted for us to be able to add new experiment types

### PR DESCRIPTION
**Pull request by dperl-dls**
Fixes #367 

Looks up the experiment plan from a "registry" which can later be made more dynamic

### To test:
1. Consider if this is a good way to approach this?
2. Run tests

### CODE REVIEW

**Comment by DominicOram** [comment]
@callumforrester and @abbiemery this will basically be replaced by `blueapi` soon, is there any way we should change this to make that move easier?

**Comment by callumforrester** [comment]
@DominicOram Probably not, current thinking for blueapi is to have a module full of plans and build the registry automagically from that.

**Review by DominicOram — COMMENTED** [review]
I think the registry idea is reasonably sensible and this will work. However, I think we could refactor around the stop a bit. Currently you need to put to `/plan/stop`, but when we move to multiple plans we probably don't want to have to keep track of which plans are running and so would be good to just be able to put to a top level `stop` endpoint. Status is also maybe a thing that you might want universal too, like a `/status` endpoint you can read from.

File: src/artemis/__main__.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ class Status(Enum):
 
 @dataclass
 class Command:
+    experiment: str
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/artemis/__main__.py:38]: yeah makes sense, will do that as well as for stop/status as mentioned in your comment

**Review by DominicOram — CHANGES_REQUESTED** [review]
If I send a bad request it's not obvious why it failed e.g.
```
(.venv) [ffv81422@ws455 python-artemis]$ curl -X PUT http://127.0.0.1:5005/bad_plan/start --data-binary "@test_parameters.json" -H "Content-Type: application/json"
{
    "status": "Success",
    "message": ""
}
(.venv) [ffv81422@ws455 python-artemis]$ curl http://127.0.0.1:5005/status
{
    "status": "Failed",
    "message": ""
}
```
I think it would be better if the `PUT` returned a `Failed` status if the plan doesn't exist? Could we have test covering this too?

**Review by DominicOram — APPROVED** [review]
Great, thank you very much!

**Pull request merged**